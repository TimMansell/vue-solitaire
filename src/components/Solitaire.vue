<template>
  <div class="solitaire">
    <Board />
    <GameOverlay v-if="!hasMoves && isGameWon" data-test="game-won">
      Congratulations, you win!
    </GameOverlay>
    <GameOverlay v-if="!hasMoves && !isGameWon" data-test="game-lost">
      Sorry, no more Moves!
    </GameOverlay>
    <Rules />
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import Board from '@/components/Board.vue';
import GameOverlay from '@/components/GameOverlay.vue';
import Rules from '@/components/Rules.vue';
// import aces from '../../tests/fixtures/boards/doubleClickAce1.json';

export default {
  name: 'Home',
  components: {
    Board,
    GameOverlay,
    Rules,
  },
  computed: {
    ...mapGetters(['isGameWon', 'hasMoves']),
  },
  mounted() {
    this.initGame();

    if (process.env.NODE_ENV === 'development') {
      // this.$store.dispatch('setBoardAndFoundation', aces);
    }
  },
  methods: {
    ...mapActions(['initGame']),
  },
};
</script>

<style scoped lang="scss">
.solitaire {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0.5rem 0;
}
</style>
