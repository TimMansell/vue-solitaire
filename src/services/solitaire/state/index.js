const setState = (state, newState) => ({ ...state, ...newState });

export default setState;
