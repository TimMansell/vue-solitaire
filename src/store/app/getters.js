const getters = {
  game: ({ game }) => game,
  isGameLoading: ({ isGameLoading }) => isGameLoading,
  isGamePaused: ({ isGamePaused }) => isGamePaused,
  isOverlayVisible: ({ isOverlayVisible }) => isOverlayVisible,
  gameOutcome: ({ gameOutcome }) => gameOutcome,
  timer: ({ game }) => game.time,
  moves: ({ game }) => game.moves.length,
  hasGameUpdated: ({ hasGameUpdated }) => hasGameUpdated,
  isLatestVersion: ({ isLatestVersion }) => isLatestVersion,
  version: ({ version }) => version,
  isConnecting: ({ connection }) => connection.isConnecting,
  isOnline: ({ connection }) => connection.isOnline,
  isCompletedGame: ({ gameOutcome }) =>
    gameOutcome.hasGameLost || gameOutcome.hasGameWon,
};

export default getters;
