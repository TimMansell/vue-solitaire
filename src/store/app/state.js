// Return state as function so we can reset it.
const state = () => ({
  game: {
    moves: [],
    time: 0,
  },
  gameOutcome: {
    hasGameWon: false,
    hasGameLost: false,
  },
  isGameLoading: true,
  isGamePaused: false,
  isOverlayVisible: true,
  versionMatch: true,
});

export default state;
