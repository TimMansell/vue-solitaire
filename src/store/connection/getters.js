const getters = {
  isConnecting: ({ isConnecting }) => isConnecting,
  hasConnectionError: ({ isConnecting, isOnline }) =>
    !isOnline && !isConnecting,
};

export default getters;
