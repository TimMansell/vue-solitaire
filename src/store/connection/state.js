const state = () => ({
  isConnecting: false,
  hasConnected: false,
  isOnline: false,
});

export default state;
