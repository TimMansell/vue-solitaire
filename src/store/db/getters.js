const getters = {
  stats: (state) => state.stats,
};

export default getters;
