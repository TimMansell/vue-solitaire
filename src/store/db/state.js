// Return state as function so we can reset it.
const state = () => ({
  userStats: {
    totalGames: null,
  },
  game: {
    id: null,
    moves: 0,
  },
});

export default state;
