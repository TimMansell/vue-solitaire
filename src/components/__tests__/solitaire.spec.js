import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Solitaire from '@/components/Solitaire.vue';
import state from '@/store/solitaire/state';
import getters from '@/store/solitaire/getters';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Solitaire.vue', () => {
  let store;

  const actions = {
    initGame: jest.fn(),
  };

  beforeEach(() => {
    store = new Vuex.Store({
      state,
      actions,
      getters,
    });
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(Solitaire, { store, localVue });

    expect(wrapper).toMatchSnapshot();
  });
});
