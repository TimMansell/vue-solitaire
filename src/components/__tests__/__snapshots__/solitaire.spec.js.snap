// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solitaire.vue matches snapshot 1`] = `
<div class="solitaire">
  <board-stub></board-stub>
  <!---->
  <rules-stub></rules-stub>
</div>
`;
